msal/src/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -43,29 +43,62 @@ namespace Microsoft.Identity.Client
-        private readonly MsalIdTokenCacheItem _msalIdTokenCacheItem;
-
+        private const string Oauth2AuthorizationHeader = "Bearer ";
+
+        /// <summary>
+        /// Constructor meant for test purposes, to allow mocking of authentication flows. Developers should 
+        /// never new up <see cref="AuthenticationResult"/> in product code.
+        /// </summary>
+        public AuthenticationResult(
+            string accessToken, 
+            bool isExtendedLifeTimeToken, 
+            string uniqueId, 
+            DateTimeOffset expiresOn, 
+            DateTimeOffset extendedExpiresOn, 
+            string tenantId, 
+            IAccount account, 
+            string idToken, 
+            IEnumerable<string> scopes)
+        {
+            AccessToken = accessToken;
+            IsExtendedLifeTimeToken = isExtendedLifeTimeToken;
+            UniqueId = uniqueId;
+            ExpiresOn = expiresOn;
+            ExtendedExpiresOn = extendedExpiresOn;
+            TenantId = tenantId;
+            Account = account;
+            IdToken = idToken;
+            Scopes = scopes;
+        }
+
         internal AuthenticationResult()
         {
         }
 
         internal AuthenticationResult(MsalAccessTokenCacheItem msalAccessTokenCacheItem, MsalIdTokenCacheItem msalIdTokenCacheItem)
         {
-            _msalAccessTokenCacheItem = msalAccessTokenCacheItem;
-            _msalIdTokenCacheItem = msalIdTokenCacheItem;
-            if (_msalAccessTokenCacheItem.HomeAccountId != null)
+            if (msalAccessTokenCacheItem.HomeAccountId != null)
             {
-                Account = new Account(_msalAccessTokenCacheItem.HomeAccountId,
-                    _msalIdTokenCacheItem?.IdToken?.PreferredUsername, _msalAccessTokenCacheItem.Environment);
+                this.Account = new Account(
+                    msalAccessTokenCacheItem.HomeAccountId,
+                    msalIdTokenCacheItem?.IdToken?.PreferredUsername,
+                    msalAccessTokenCacheItem.Environment);
             }
+
+            this.AccessToken = msalAccessTokenCacheItem.Secret;
+            this.UniqueId = msalIdTokenCacheItem?.IdToken?.GetUniqueId();
+            this.ExpiresOn = msalAccessTokenCacheItem.ExpiresOn;
+            this.ExtendedExpiresOn = msalAccessTokenCacheItem.ExtendedExpiresOn;
+            this.TenantId = msalIdTokenCacheItem?.IdToken?.TenantId;
+            this.IdToken = msalIdTokenCacheItem?.Secret;
+            this.Scopes = msalAccessTokenCacheItem.ScopeSet;
+            this.IsExtendedLifeTimeToken = msalAccessTokenCacheItem.IsExtendedLifeTimeToken;
         }
 
         /// <summary>
-        /// Gets the Access Token to use as a bearer token to access the protected web API
+        /// Access Token that can be used as a bearer token to access protected web APIs
         /// </summary>
-        public virtual string AccessToken => _msalAccessTokenCacheItem.Secret;
```

Comment:
I fail to see the point in having virtual properties on a data class. Developers cannot inherit from it anyway, because there is no public constructor (you get compilation errors). We never derive from it.